tasks/Task.cpp

```diff
@@ -12,44 +12,93 @@
 #include <iostream>
 #include <json/json.h>
 
+#include <list>
+
 using namespace controldev_websocket;
 using namespace controldev;
 using namespace seasocks;
 
-struct controldev_websocket::JoystickHandler : WebSocket::Handler {
+struct controldev_websocket::Client {
     WebSocket *connection = nullptr;
+    std::string id = "";
+    Json::FastWriter fast;
+    bool is_new = true;
+
+    bool isEmpty(){
+        return connection == nullptr;
+    }
+
+    void steal(Client &other){
+        if (other.isEmpty() and !is_new){
+            return;
+        }
+        Json::Value feedback;
+        if (other.isEmpty()){
+            feedback["connection_state"]["state"] = "new";
+        } else {
+            feedback["connection_state"]["state"] = "lost";
+            feedback["connection_state"]["peer"] = id.empty() ? "pending connection" : id;
+            other.connection->send(fast.write(feedback));
+            other.connection->close();
+            feedback["connection_state"]["state"] = "stolen";
+            feedback["connection_state"]["peer"] = other.id.empty() ? "pending connection"
+                                                                      : other.id;
+        }
+        connection->send(fast.write(feedback));
+        is_new = false;
+    };
+};
+struct controldev_websocket::JoystickHandler : WebSocket::Handler {
+    Client pending;
+    Client controlling;
     Task *task = nullptr;
     Json::Value msg;
     Json::FastWriter fast;
     Statistics statistic;
 
     void onConnect(WebSocket *socket) override{
-        if (connection != nullptr){
-            connection->close();
-        }
-        connection = socket;
-        task->is_controlling = false;
+        Client new_pending;
+        new_pending.connection = socket;
+        new_pending.steal(pending);
+        pending = new_pending;
     }
     void onData(WebSocket *socket, const char *data) override{
-        if (connection != socket) {
+
+        if (socket != pending.connection and socket != controlling.connection) {
             LOG_WARN_S << "Received message from inactive connection" << std::endl;
             return;
         }
-
-        ++task->received;
-        bool result = task->handleIncomingWebsocketMessage(data, socket);
+        bool result = task->parseIncomingWebsocketMessage(data, socket);
```

Comment:
OK. I'm officially stressed out by the "behind the scene" passing of decoded JSON data. This is way too much implicit dependency between the call to `parse` and the call to `handle`

Make the JSON data a return value of the method, and pass it as argument to other methods.